mlflow/store/tracking/file_store.py

```diff
@@ -1274,6 +1287,115 @@ class _SummaryTuple:
+        self.set_tag(run_id, RunTag(MLFLOW_RUN_SOURCE_TYPE, "PROMPT_ENGINEERING"))
+
+        artifact_dir = self._get_artifact_dir(experiment_id, run_id)
+
+        # log model
+        from mlflow.models import Model
+
+        utc_time_created = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S.%f")
+        promptlab_model = Model(
+            artifact_path=artifact_dir,
+            run_id=run_id,
+            utc_time_created=utc_time_created,
+        )
+        self.record_logged_model(run_id, promptlab_model)
+
+        # write artifact files
+        from mlflow.store.artifact.artifact_repository_registry import get_artifact_repository
+
+        artifact_repo = get_artifact_repository(artifact_dir)
+
+        local_dir = tempfile.mkdtemp()
+        model_json = create_model_file(
+            run_id, mlflow_version, prompt_parameters, promptlab_model.model_uuid, utc_time_created
+        )
+        model_json_file_path = os.path.join(local_dir, "model", "MLmodel")
+        make_containing_dirs(model_json_file_path)
+        write_to(model_json_file_path, model_json)
+
+        conda_yaml = create_conda_yaml_file(mlflow_version)
+        conda_yaml_file_path = os.path.join(local_dir, "model", "conda.yaml")
+        make_containing_dirs(conda_yaml_file_path)
+        write_to(conda_yaml_file_path, conda_yaml)
+
+        python_yaml = create_python_env_file()
+        python_yaml_file_path = os.path.join(local_dir, "model", "python_env.yaml")
+        make_containing_dirs(python_yaml_file_path)
+        write_to(python_yaml_file_path, python_yaml)
+
+        requirements_txt = create_requirements_txt_file(mlflow_version)
+        requirements_txt_file_path = os.path.join(local_dir, "model", "requirements.txt")
+        make_containing_dirs(requirements_txt_file_path)
+        write_to(requirements_txt_file_path, requirements_txt)
+
+        loader_module = create_loader_file(
+            prompt_parameters, prompt_template, model_parameters, model_route
+        )
+        loader_module_file_path = os.path.join(
+            local_dir, "model", "loader", "gateway_loader_module.py"
+        )
+        make_containing_dirs(loader_module_file_path)
+        write_to(loader_module_file_path, loader_module)
+
+        eval_results_json = create_eval_results_file(
+            prompt_parameters, model_input, model_output_parameters, model_output
+        )
+        eval_results_json_file_path = os.path.join(local_dir, "eval_results_table.json")
+        make_containing_dirs(eval_results_json_file_path)
+        write_to(eval_results_json_file_path, eval_results_json)
+
+        input_example_json = create_input_example_file(prompt_parameters)
+        input_example_json_file_path = os.path.join(local_dir, "input_example.json")
+        make_containing_dirs(input_example_json_file_path)
+        write_to(input_example_json_file_path, input_example_json)
+
+        artifact_repo.log_artifacts(local_dir)
+        shutil.rmtree(local_dir)
```

Comment:
/ai Explain why we should use `with tempfile.TemporaryDirectory():` instead of `tempfile.mkdtemp()` and `shutil.rmtree(local_dir)`.